src/zh/manhuaren/src/eu/kanade/tachiyomi/extension/zh/manhuaren/RemoveTokenInterceptor.kt

```diff
@@ -0,0 +1,38 @@
+package eu.kanade.tachiyomi.extension.zh.manhuaren
+
+import android.content.SharedPreferences
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.json.JSONObject
+import java.io.IOException
+
+class RemoveTokenInterceptor(private val baseUrl: String, private val preferences: SharedPreferences) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        if (!request.url.toString().startsWith(baseUrl)) return chain.proceed(request)
+
+        var response: Response? = null
+        val failedResponse = try {
+            response = chain.proceed(request)
+            if (response.isSuccessful) return response
+            response.close()
+            Result.success(response)
+        } catch (e: Throwable) {
+            if (chain.call().isCanceled()) throw e
+            Result.failure(e)
+        }
+
+        if (response != null && JSONObject(response.body.string()).has("errorResponse")) {
```

Comment:
Consider using `kotlinx.serialization` instead of `org.json`.

```suggestion
        if (response != null && JSONObject(response.body.string()).has("errorResponse")) {
```